src/lib/modernExtend.ts

```diff
@@ -466,17 +466,29 @@ export interface OnOffArgs {
     skipDuplicateTransaction?: boolean;
     endpointNames?: string[];
     configureReporting?: boolean;
+    description?: string;
 }
 export function onOff(args?: OnOffArgs): ModernExtend {
     args = {powerOnBehavior: true, skipDuplicateTransaction: false, configureReporting: true, ...args};
 
-    const exposes: Expose[] = exposeEndpoints(e.switch(), args.endpointNames);
+    const exposes: Expose[] = args.description
+        ? exposeEndpoints(e.switch(args.description), args.endpointNames)
+        : exposeEndpoints(e.switch(), args.endpointNames);
 
     const fromZigbee: Fz.Converter[] = [args.skipDuplicateTransaction ? fz.on_off_skip_duplicate_transaction : fz.on_off];
     const toZigbee: Tz.Converter[] = [tz.on_off];
 
     if (args.powerOnBehavior) {
-        exposes.push(e.power_on_behavior(['off', 'on', 'toggle', 'previous']));
+        const powerOnBehaviorExpose = args.description
+            ? e.power_on_behavior(['off', 'on', 'toggle', 'previous']).withDescription(args.description)
```

Comment:
Please add a separate description for powerOnBehavior, change `powerOnBehavior?: boolean;` to `powerOnBehavior?: boolean | {description: string};`

Reply:
ok